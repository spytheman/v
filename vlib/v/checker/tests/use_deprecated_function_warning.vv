[deprecated]
fn xyz() {
	println('hi')
}

[deprecated: 'use foo2 instead']
fn abc() {
	println('hi')
}

struct Abc {}
[deprecated: 'please use a.xyz()']
fn (a Abc) method() {
	println('hi from ${@METHOD}')
}

fn main() {
	xyz()
	abc()
	a := Abc{}
	a.method()
}
