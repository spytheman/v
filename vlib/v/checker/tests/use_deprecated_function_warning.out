vlib/v/checker/tests/use_deprecated_function_warning.vv:18:2: error: function `xyz` has been deprecated
   16 | 
   17 | fn main() {
   18 |     xyz()
      |     ~~~~~
   19 |     abc()
   20 |     a := Abc{}
vlib/v/checker/tests/use_deprecated_function_warning.vv:19:2: error: function `abc` has been deprecated; use foo2 instead
   17 | fn main() {
   18 |     xyz()
   19 |     abc()
      |     ~~~~~
   20 |     a := Abc{}
   21 |     a.method()
vlib/v/checker/tests/use_deprecated_function_warning.vv:21:4: error: method `method` has been deprecated; please use a.xyz()
   19 |     abc()
   20 |     a := Abc{}
   21 |     a.method()
      |       ~~~~~~~~
   22 | }
